File: app/src/main/java/igrek/todotree/services/tree/persistence/ItemAttribute.java
package igrek.todotree.services.tree.persistence;


class ItemAttribute {
	
	private String name;
	private String value;
	
	ItemAttribute(String name, String value) {
		this.name = name;
		this.value = value;
	}
	
	public String getName() {
		return name;
	}
	
	public String getValue() {
		return value;
	}
	
	@Override
	public boolean equals(Object obj) {
		if (!(obj instanceof ItemAttribute))
			return false;
		ItemAttribute attr2 = (ItemAttribute) obj;
		return name.equals(attr2.name) && value.equals(attr2.value);
	}
	
	@Override
	public String toString() {
		return name + " = " + value;
	}
}

File: app/src/main/java/igrek/todotree/service/tree/persistence/ItemAttribute.kt
package igrek.todotree.service.tree.persistence

internal class ItemAttribute(val name: String, val value: String) {

    override fun equals(other: Any?): Boolean {
        if (other !is ItemAttribute) return false
        return name == other.name && value == other.value
    }

    override fun toString(): String {
        return "$name = $value"
    }

    override fun hashCode(): Int {
        var result = name.hashCode()
        result = 31 * result + value.hashCode()
        return result
    }
}